test/integration/noderesourcetopology_test.go

```diff
@@ -0,0 +1,636 @@
+			Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
+				{
+					Name: "v1alpha1",
+					Schema: &apiextensionsv1.CustomResourceValidation{
+						OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+							Type: "object",
+							Properties: map[string]apiextensionsv1.JSONSchemaProps{
+								"topologyPolicies": {
+									Type: "array",
+									Items: &apiextensionsv1.JSONSchemaPropsOrArray{
+										Schema: &apiextensionsv1.JSONSchemaProps{
+											Type: "string",
+										},
+									},
+								},
+								"zones": {
+									Type: "array",
+									Items: &apiextensionsv1.JSONSchemaPropsOrArray{
+										Schema: &apiextensionsv1.JSONSchemaProps{
+											Type: "object",
+											Properties: map[string]apiextensionsv1.JSONSchemaProps{
+												"name": {
+													Type: "string",
+												},
+												"type": {
+													Type: "string",
+												},
+												"parent": {
+													Type: "string",
+												},
+												"resources": {
+													Type: "array",
+													Items: &apiextensionsv1.JSONSchemaPropsOrArray{
+														Schema: &apiextensionsv1.JSONSchemaProps{
+															Type: "object",
+															Properties: map[string]apiextensionsv1.JSONSchemaProps{
+																"name": {
+																	Type: "string",
+																},
+																"capacity": {
+																	XIntOrString: true,
+																},
+																"allocatable": {
+																	XIntOrString: true,
+																},
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					Served:  true,
+					Storage: true,
+				},
+			},
+		},
+	}
+}
+
+func createNodeResourceTopologies(ctx context.Context, topologyClient *topologyclientset.Clientset, noderesourcetopologies []*topologyv1alpha1.NodeResourceTopology) error {
+	for _, nrt := range noderesourcetopologies {
+		_, err := topologyClient.TopologyV1alpha1().NodeResourceTopologies(nrt.Namespace).Create(ctx, nrt, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func cleanupNodeResourceTopologies(ctx context.Context, topologyClient *topologyclientset.Clientset, noderesourcetopologies []*topologyv1alpha1.NodeResourceTopology) {
+	for _, nrt := range noderesourcetopologies {
+		err := topologyClient.TopologyV1alpha1().NodeResourceTopologies(nrt.Namespace).Delete(ctx, nrt.Name, metav1.DeleteOptions{})
+		if err != nil {
+			klog.Errorf("clean up NodeResourceTopologies (%v/%v) error %s", nrt.Namespace, nrt.Name, err.Error())
+		}
+	}
+}
+
+func withContainer(pod *v1.Pod, image string) *v1.Pod {
+	pod.Spec.Containers[0].Name = "con0"
+	pod.Spec.Containers[0].Image = image
+	return pod
+}
+
+// Req adds a new container to the inner pod with given resource map.
+func Req(p *st.PodWrapper, resMap map[v1.ResourceName]string) *st.PodWrapper {
```

Comment:
to make the names consistent, `withReqAndLimit()` might be more descriptive as you set both reqs and limits.